Bluetooth media player messaging

aselafernando opened the pull request:
Created messaging from the Bluetooth media player. This will broadcast the current track details and track position so plugins can make use of this. This will allow writing this data to a vehicle display over CANBUS or serial etc.

aselafernando committed:
Added media player callbacks

aselafernando committed:
Added messaging for track and position

aselafernando committed:
Changed case for messages, added better position handling

aselafernando committed:
Merge branch 'viktorgino:master' into PhoneBluetooth

viktorgino commented:
I think it'd be a better idea to a media position and media info changed signal to [MediaInterface](https://github.com/viktorgino/headunit-desktop/blob/master/includes/mediainterface.h), so all the other media plugins can use it.
Then [MediaManager](https://github.com/viktorgino/headunit-desktop/blob/master/app/mediamanager.h) could handle sending it onto the messaging bus etc

aselafernando committed:
MediaManager now places track position and track information on the message bus

aselafernando commented:
Yep that makes sense, I have modifed the code so now MediaManager is the interface to the message bus.

```MediaManager::mediaPosition quint32``` and ```MediaManager::mediaTrack QVariantMap``` are the two messages. Just tested this and it worked well. Could probably drop "media" from the id as well.

aselafernando committed:
Dropped "media" from message id

aselafernando committed:
Changed MediaManager to MediaInput message sender

aselafernando commented:
I changed the messages to ```MediaInput::position quint32``` and ```MediaInput::track QVariantMap``` to be more consistent with the existing

aselafernando committed:
Merge branch 'viktorgino:master' into PhoneBluetooth

aselafernando committed:
Merge branch 'viktorgino:master' into PhoneBluetooth

aselafernando committed:
Fixed issues with playing/pausing with AA

TelephonyManager::onMediaStatus() when raising playbackStarted was not checking if AA was connected.

TelephonyManager::mediaPlaybackStarted() and TelephonyManager::onMediaStatus() both raise playbackStarted();

TelephonyManager::mediaPlaybackStarted() might be redundant. Needs to be removed from Phone.qml in a future change.

aselafernando committed:
Added bluetooth reconnection timer

This allows bluetooth connections to be made later if the device is not present when the adaptor is initialised. Similarly, they can be remade if the connection is lost.

aselafernando committed:
Fixed bug preventing bluetooth retries

aselafernando committed:
Changed approach to reconnecting

aselafernando committed:
Stopped mediaPlayer signals accumulating

Old connections were not disconnected when a media connection was reestablished, leading to triggers of signals/slots

aselafernando committed:
FIXED: MediaManager sending multiple events

MediaManager was connected to the message handler on each plugin load. This caused every message from MediaManager to be repeated based on the number of plugins loaded.
